### DISCUSSION TITLE
How to get the first and last note startTick?

### DISCUSSION THREAD

**Discussion by renanboni** [CoderLine/alphaTab#1190] [Q&A] [answered]
hey! hope you're doing great :)

I'm building an app that uses **alphatab** as a WebView but the problem is that I can't select the playback range (when I tap and hold to select the range the web view selects the text instead and it doesn't seem to pass down the event to **alphatab**) and a workaround would be to create a range where the user could slide to select a range and I would manually call the alphatab api.

I did most of the logic but I'm struggling to get the **startTick** of the first and last notes so I can build up my range.

Here's what I have so far:

```
function getRange(score, trackIndex) {
    // Ensure the track index is within range
    if (score.tracks.length <= trackIndex) {
        throw new Error('Invalid track index');
    }

    let displayStartList = [];

    // Traverse the object hierarchy
    let track = score.tracks[trackIndex];
    for (let staff of track.staves) {
        for (let bar of staff.bars) {
            for (let voice of bar.voices) {
                for (let beat of voice.beats) {
                    // Add the displayStart property to the list
                    displayStartList.push(beat.displayStart);
                }
            }
        }
    }

    return displayStartList;
}

```
// then I would call
```
function setPlaybackRangePosition(start, end) {
    this.api.playbackRange = { startTick: start, endTick: end };
}
```

but it doesn't seem to be accurate as some of the last notes are left behind, I've also tried with **absoluteDisplayStart** and **absolutePlaybackStart** and I got similar result.

is there a way to do what I've been trying?

thank you

**Comment by Danielku15** [comment, marked as answer]
I think there are two ways to tackle this topic: 

1. Ensure/Implement proper touch input support. You could check out how the normal interaction with mouse events is realized [here](https://github.com/CoderLine/alphaTab/blob/10e40f4b9749b028ab158b8379a87c59e18872ad/src/AlphaTabApiBase.ts#L1225-L1297) and implement a touch compatible counterpart for your needs. The API should be public enough to implement a custom touch variant fitting your needs (tap, tap-hold etc.). 
2. Use some workaround like you propose and use the `MidiTickLookup` to lookup the correct range like described here: https://github.com/CoderLine/alphaTab/discussions/1126
